gossip/c_block_callbacks.go

```diff
@@ -589,6 +597,47 @@ func consensusCallbackBeginBlockFn(
 	}
 }
 
+// rlpEncodedMaxHeaderSizeInBytes is an upper bound of the EVM block header size
+// used for block size calculations.
+const rlpEncodedMaxHeaderSizeInBytes = 1024
+
+// processUserTransactions executes user transactions in order, adding them to the block
+// until all transactions are processed or the gas/block size limit is reached.
+func processUserTransactions(evmProcessor blockproc.EVMProcessor, blockBuilder *inter.BlockBuilder, orderedTxs []*types.Transaction, userTransactionGasLimit uint64) {
+	remainingGas := userTransactionGasLimit
+	remainingSize := uint64(params.MaxBlockSize - rlpEncodedMaxHeaderSizeInBytes)
+	internalTxs := blockBuilder.GetTransactions()
+	for _, tx := range internalTxs {
+		remainingSize -= tx.Size()
+	}
+	for _, tx := range orderedTxs {
+		neededSpace := txSizeIncSubsidies(tx)
+		if neededSpace < remainingSize && tx.Gas() < remainingGas {
+			for _, processed := range evmProcessor.Execute([]*types.Transaction{tx}, remainingGas) {
+				if processed.Receipt != nil { // < nil if skipped
+					blockBuilder.AddTransaction(
+						processed.Transaction,
+						processed.Receipt,
+					)
+					remainingGas = userTransactionGasLimit - processed.Receipt.CumulativeGasUsed
+					remainingSize -= txSizeIncSubsidies(tx)
+				}
+			}
+		}
+	}
+}
+
+// txSizeIncSubsidies returns the size of the transaction,
+// including any overhead introduced by sponsorship requests.
+func txSizeIncSubsidies(tx *types.Transaction) uint64 {
```

Comment:
maybe this is just me, but `Inc` could be confused with `Increment` instead of `Include`. what about `txSizeWithSubsidies` ?